src/ccl_cls.c

```diff
@@ -844,16 +854,16 @@ static double cl_integrand(double lk,void *params)
 //clt2 -> tracer #2
 //l    -> angular multipole
 //lkmin, lkmax -> log of the range of scales where the transfer functions have support
-static void get_k_interval(ccl_cosmology *cosmo,CCL_ClWorkspace *w,
-			   CCL_ClTracer *clt1,CCL_ClTracer *clt2,int l,
+static void get_k_interval(ccl_cosmology *cosmo,
+			   CCL_ClTracer *clt1,CCL_ClTracer *clt2,double l,
 			   double *lkmin,double *lkmax)
 {
   double chimin,chimax;
   int cut_low_1=0,cut_low_2=0;
   
   //Define a minimum distance only if no lensing is needed
-  if((clt1->tracer_type==ccl_number_counts_tracer) && (clt1->has_magnification==0)) cut_low_1=1;
-  if((clt2->tracer_type==ccl_number_counts_tracer) && (clt2->has_magnification==0)) cut_low_2=1;
+  if((clt1->has_shear==0) && (clt1->has_magnification==0)) cut_low_1=1;
+  if((clt2->has_shear==0) && (clt2->has_magnification==0)) cut_low_2=1;
```

Comment:
yep, that's fine. This is here to limit the range of integration for those tracers that are localized in redshifts. An IA-only tracer or a number counts tracer without magnification both fit the bill. This checks for the presence of terms that would make the transfer functions non-local in redshift.

In case you're wondering, these will also catch CMB lensing tracers as if they were localized, but that's fine, because their values for chi_min and chi_max are set up correctly.

Reply:
I was not wondering about the CMB lensing tracers. Can you add comments in the code with the info above?

Reply:
Done!